src/main/kotlin/domain/geometric/Dimension.kt

```diff
@@ -0,0 +1,35 @@
+package domain.geometric
+
+data class Dimension(
+    val width: Int,
+    val height: Int,
+) {
+    val area: Int = width * height
+
+    init {
+        require(width > 0 && height > 0) {
+            "너비와 높이는 모두 0보다 큰 값이어야 합니다. 입력된 너비 = $width, 입력된 높이 = $height"
+        }
+    }
+
+    fun fill(): List<Location> {
+        val rows = (0 until height).toList()
+        val columns = (0 until width).toList()
+        return rows.flatMap { row ->
+            mapToLocation(columns, row)
+        }
+    }
+
+    private fun mapToLocation(
+        columns: List<Int>,
+        row: Int
+    ): List<Location> = columns.map { col ->
+        Location(LocationValue(row), LocationValue(col))
+    }
```

Comment:
이 부분에 약간의 코틀린 감성을 섞어본다면 아래와 같이 확장함수를 이용해서 표현할수도 있을것 같네요~!

```suggestion
    fun fill(): List<Location> {
        val rows = (0 until height).toList()
        val columns = (0 until width).toList()
        return rows.flatMap { row -> columns.mapToLocation(row) }
    }

    private fun List<Int>.mapToLocation(row: Int): List<Location> =
        this.map { column -> Location(LocationValue(row), LocationValue(column)) }

```